pkg/reconciler/instances/istio/webhooks/webhooks.go

```diff
@@ -0,0 +1,82 @@
+package webhooks
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	retry "github.com/avast/retry-go"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/instances/istio/clientset"
+	"go.uber.org/zap"
+	"istio.io/istio/istioctl/pkg/tag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	retriesCount        = 5
+	delayBetweenRetries = 5 * time.Second
+)
+
+var deactivatedLabel = map[string]string{
+	"istio.io/deactivated": "never-match",
+}
+
+// DeleteConflictedDefaultTag deletes conflicted tagged MutatingWebhookConfiguration, if it exists and if the default revision MutatingWebhookConfiguration is not deactivated by istioctl.
+func DeleteConflictedDefaultTag(ctx context.Context, provider clientset.Provider, kubeConfig string, logger *zap.SugaredLogger) error {
+	kubeClient, err := provider.RetrieveFrom(kubeConfig, logger)
+	if err != nil {
+		logger.Error("Could not retrieve KubeClient from Kubeconfig!")
+		return err
+	}
+
+	retryOpts := []retry.Option{
+		retry.Delay(delayBetweenRetries),
+		retry.Attempts(uint(retriesCount)),
+		retry.DelayType(retry.FixedDelay),
+	}
+
+	err = retry.Do(func() error {
+		webhooks, err := tag.GetWebhooksWithTag(ctx, kubeClient, tag.DefaultRevisionName)
+		if err != nil {
+			return err
+		}
+
+		if !isDefaultRevisionDeactivated(ctx, kubeClient) && len(webhooks) > 0 {
+			for _, wh := range webhooks {
+				err := kubeClient.AdmissionregistrationV1().MutatingWebhookConfigurations().Delete(ctx, wh.Name, metav1.DeleteOptions{})
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		return nil
+	}, retryOpts...)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func isDefaultRevisionDeactivated(ctx context.Context, client kubernetes.Interface) bool {
+	webhooks, err := tag.GetWebhooksWithRevision(ctx, client, tag.DefaultRevisionName)
+	if err != nil {
+		return true
+	}
+	if len(webhooks) == 0 {
+		return true
+	}
+
+	webhook := webhooks[0]
```

Comment:
Why can we rely on the order of the webhooks and always check the first webhook? As far as I understand this `GetWebhooksWithRevision` will return `istio-sidecar-injector` and `istio-revision-tag-default` and we only want to verify that all webhooks in the `istio-sidecar-injector` webhook configuration are deactivated. Should we add a filter for the name to make it more clear?